### PR TITLE
SASS-3172 - Create of State Benefits FE microservice.

### PR DESCRIPTION
**Pull request by georgievh**

### CODE REVIEW

**Comment by georgievh** [comment]
test this please

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import play.api.i18n.Lang
+import play.api.mvc.{Call, RequestHeader}
+import uk.gov.hmrc.play.bootstrap.binders.SafeRedirectUrl
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class AppConfig @Inject()(servicesConfig: ServicesConfig) {
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/config/AppConfig.scala:27]: All that are added are actually needed. They are referred by the common templates.

File: app/config/ErrorHandler.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import play.api.mvc.Results._
+import play.api.mvc.{Request, RequestHeader, Result}
+import play.twirl.api.Html
+import uk.gov.hmrc.play.bootstrap.frontend.http.FrontendErrorHandler
+import views.html.templates.{InternalServerErrorTemplate, NotFoundTemplate, ServiceUnavailableTemplate}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.Future
+
+// TODO: Test me
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/config/ErrorHandler.scala:30]: Our integration test base is different and it is not just copying over the tests.

File: app/controllers/SessionExpiredController.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import models.authorisation.SessionValues
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.SessionHelper
+import views.html.templates.TimeoutPage
+
+import javax.inject.Inject
+
+
+class SessionExpiredController @Inject()(pageView: TimeoutPage)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/SessionExpiredController.scala:29]: Same as above

File: app/controllers/errors/AgentAuthErrorController.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+package controllers.errors
+
+import config.AppConfig
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.templates.AgentAuthErrorPageView
+
+import javax.inject.Inject
+
+class AgentAuthErrorController @Inject()(pageView: AgentAuthErrorPageView)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/errors/AgentAuthErrorController.scala:26]: Same as above.

File: app/controllers/errors/IndividualAuthErrorController.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+package controllers.errors
+
+import config.AppConfig
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.templates.IndividualUnauthorisedView
+
+import javax.inject.Inject
+
+class IndividualAuthErrorController @Inject()(pageView: IndividualUnauthorisedView)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/errors/IndividualAuthErrorController.scala:26]: Same as above, but I also wrote unit tests for them

File: app/controllers/errors/TaxYearErrorController.scala [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import config.AppConfig
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{SessionHelper, TaxYearHelper}
+import views.html.templates.TaxYearErrorTemplate
+
+import javax.inject.Inject
+
+class TaxYearErrorController @Inject()(actionsProvider: ActionsProvider,
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/errors/TaxYearErrorController.scala:29]: Same as above.

File: app/controllers/errors/UnauthorisedUserErrorController.scala [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+package controllers.errors
+
+import config.AppConfig
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.templates.UnauthorisedUserErrorPageView
+
+import javax.inject.Inject
+
+class UnauthorisedUserErrorController @Inject()(pageView: UnauthorisedUserErrorPageView)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/errors/UnauthorisedUserErrorController.scala:26]: Same as above.

File: app/controllers/errors/YouNeedAgentServicesController.scala [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+package controllers.errors
+
+import config.AppConfig
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.templates.YouNeedAgentServicesView
+
+import javax.inject.Inject
+
+class YouNeedAgentServicesController @Inject()(pageView: YouNeedAgentServicesView)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply app/controllers/errors/YouNeedAgentServicesController.scala:26]: Same as above.

**Review by jhkldev — APPROVED** [review]

**Pull request merged**